pilot/proxy/envoy/config.go

```diff
@@ -886,10 +886,10 @@ func buildEgressHTTPRoutes(mesh *meshconfig.MeshConfig, node proxy.Node,
 			if !model.IsEgressRulesSupportedHTTPProtocol(protocol) {
 				continue
 			}
-			intPort := int(port.Port)
-			modelPort := &model.Port{Name: fmt.Sprintf("external-%v-%d", protocol, intPort),
-				Port: intPort, Protocol: protocol}
-			httpConfig := httpConfigs.EnsurePort(intPort)
+			portNumber := int(port.GetNumber())
+			modelPort := &model.Port{Name: fmt.Sprintf("external-%v-%d", protocol, portNumber),
+				Port: portNumber, Protocol: protocol}
+			httpConfig := httpConfigs.EnsurePort(portNumber)
```

Comment:
See my earlier comment. We have a Port struct in the service model. Just convert the Port (proto) to that struct. That would eliminate these changes. you could even hoist the port name construction (external-x-y) into that conversion function..

@ijsnellf thoughts? I leave it to you guys